_posts/2017-11-21-Installing-Jenkins-with-PowerShell-DSC.md

```diff
@@ -0,0 +1,745 @@
+---
+layout: post
+title: Installing Jenkins with PowerShell DSC
+author: Rinorragi
+date: 2017-11-21 13:20:00 +0200
+excerpt: How to use PowerShell DSC in installing your continuous integration server.
+categories: 
+- Episerver
+tags: 
+- Episerver 
+- DOTNET 
+- DevOps
+- PowerShell
+- Jenkins
+---
+I have been struggling to find a good reference about how to setup a Jenkins environment in Microsoft environment with automatic installation script. So I decided to write a blog post about it. I also want to write about how to do continuous integration with Episerver DXC but felt that I need to first tell how to setup the Jenkins. Our aim is to put up a Jenkins server that can build .NET FrameWork MVC application and front end. If you are not a reader type of a person, then [here](https://github.com/solita/powershell-dsc-jenkins) is a GitHub link to example scripts.
+
+## About the PowerShell DSC
+
+You might not be familiar with the PowerShell DSC and it comes from Desired Stage Configuration. It has been there now for years still in my experience most .NET developers are unfamiliar with it. DSC exists for various reasons:
+
+* Make scripting less complex
+* Make scripting to look the same for smaller learning curve 
+* Make scripting pieces to be more reusable
+* Make installation scripts to be idempotent (repeatable)
+
+The DSC is all about the setting the state of a machine to be certain. Most used example is to make sure that a service is running or that specific file is found on given location. If you dig further on to this world you will find concepts of pull and push servers that would help you to set a farm of machines into certain state. We will not use those but we run the script locally with the help of LCM which is "local configuration manager". If you are looking for basics of PowerShell Desired State Configuration then this [blog](https://red-gate.com/simple-talk/sysadmin/powershell/powershell-desired-state-configuration-the-basics/) was a well-written one.
```

Comment:
- [ ] "Is all about **the** setting the" (poista ylimääräinen the)
- [ ] "If you dig further **on to**" -> "into"
- [ ] Voit käyttää jatkossa akronyymiä DSC, koska se on selitetty auki aiempana. Muuta kaikki "Desired State Configuration" -> DSC:ksi.